pkg/specialservices/resolver.go

```diff
@@ -40,6 +41,10 @@ type NameserverFinder struct {
 }
 
 func (f *NameserverFinder) DNSNamespace(ctx *synccontext.SyncContext) (client.Client, string) {
+	if ctx.Config.Sync.ToHost.Namespaces.Enabled {
+		// when namespaces are synced, we rely on translator logic
```

Comment:
But why? Please make this comment more specific and tell us why exactly this is needed here

Reply:
Ok found the root cause - new namespace mapper needed a fallback that would handle those cases, that's why we were getting empty namespace. Fixed and cleaned this up